### ISSUE TITLE
How to extract xy coordinates and curves

### ISSUE THREAD

**Issue by AnupCloud** [ultralytics/ultralytics#20102]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

1. I want to extract(segment) the curves
2. Then need to know the XY coordinates value for each curves

<img width="1224" alt="Image" src="https://github.com/user-attachments/assets/ef8c3497-ba6e-428e-9721-a475d48001dd" />

Result from yolov8-seg(nano) , Dataset size 66 images
<img width="629" alt="Image" src="https://github.com/user-attachments/assets/36c7ff07-321b-4689-b222-c11ff431d5b8" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AnupCloud, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) where you can explore [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. These resources might help address your query regarding extracting XY coordinates and curves.

If this is a custom training ❓ Question, please provide additional details like dataset examples, training logs, and clarify your requirements, ensuring you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

If this is a 🐛 Bug Report, please include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate the issue effectively.

Meanwhile, you can also connect with our community for further assistance:
- Join our [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time support.
- Engage in discussions on [Discourse](https://community.ultralytics.com/).
- Share insights on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Ensure you're using the latest version of the `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). Update with:

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in any of these verified environments, complete with all dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. These tests validate all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response, but rest assured, an Ultralytics engineer will review your issue and provide further assistance soon. 😊

**Comment by AnupCloud** [comment]
Hello @glenn-jocher,

Could you please explain the process in a bit more detail?
I’ve trained the model, but I’m not getting good prediction results — it seems to be overfitting.
Here’s what I’ve done:
I used the YOLOv8-n detection model along with the SAM_L model for segmentation.

The segmentation works well on the training data, but it's not performing properly on the test data.

[PFA] – please find the attachment for reference.

<img width="1601" alt="Image" src="https://github.com/user-attachments/assets/396e3433-00dc-4b52-8b21-74f09265075a" />

**Comment by AnupCloud** [comment]
Do I need to blur the grid in the image before annotation to achieve better segmentation results? Also, do I need to perform postprocessing before predicting on the images?

![Image](https://github.com/user-attachments/assets/6e3564b5-efe9-42f2-8fa6-ef4cbffbebc7)

**Comment by glenn-jocher** [comment]
Hi @AnupCloud,

Regarding your questions:

1. **Grid blurring**: Yes, blurring or removing the grid before annotation could help the model focus on the actual curves rather than the background patterns. This would reduce the visual noise that might confuse the segmentation model.

2. **Postprocessing**: You typically would apply postprocessing after prediction, not before. For your curve extraction, you might want to apply contour simplification or smoothing to the obtained coordinates. 

For example, after getting your coordinates with `masks.xy`, you could use OpenCV's `cv2.approxPolyDP()` to simplify the curves or apply spline fitting if you need smoother representations of the extracted curves.

If you're primarily interested in the curves themselves rather than perfect pixel-level segmentations, you might also consider training a direct curve detection model instead of a segmentation model.
